mycelo/config/config.go

```diff
@@ -0,0 +1,191 @@
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"math/big"
+	"os"
+
+	"github.com/ethereum/go-ethereum/internal/fileutils"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type Environment struct {
+	Paths         Paths
+	EnvConfig     EnvConfig
+	GenesisConfig GenesisConfig
+
+	// Derived Fields
+	accounts *GenesisAccounts
+}
+
+func NewEnv(envpath string, cfg *EnvConfig) (*Environment, error) {
+	env := &Environment{
+		Paths:         Paths{Workdir: envpath},
+		EnvConfig:     *cfg,
+		GenesisConfig: *BaseContractsConfig(),
+	}
+
+	accounts, err := createGenesisAccounts(cfg)
```

Comment:
this line implies we can't modify cfg after calling `NewEnv()`.

Reply:
That is correct, that requires some reworking (which explains the overrides function)